### ISSUE TITLE
the number of parameters does not match

### ISSUE THREAD

**Issue by wululu-1017** [osudrl/masked-humanoid-controller#2]
This function is defined like this：
`def visualize_keypoint_dict(keypoint_dict, mask, player, motion_id, collect_dataset = False):
    cached_vars = generate_cache_for_env_vars(player.env.task)

    load_humanoid_keypoint_data_to_sword_shield_demo_store(player, keypoint_dict)

    # import time
    # time.sleep(5)
    for _ in range(1):
        i = motion_id 
        print(list(keypoint_dict.keys())[i])
        demo_idx_pool = [i]

        player._render_demo = False
        player._render_target_markers = True

        # Reset ############################################################################################################
        player.env.task._demo_motion_weights[:] = 0
        player.env.task._demo_motion_weights[demo_idx_pool] = 1
        player.env.task._demo_motion_weights = player.env.task._demo_motion_weights / torch.sum(player.env.task._demo_motion_weights)
        # player.env.task._fixed_demo_lookahead_mask_env2idxmap_flag = True if enable_lookahead_mask else False
        # ################################################################################

        # Reset Tensors
        player.env.task._random_dir_speed_lookahead_share = 0 #cfg["env"]["random_dir_speed_lookahead_share"]

        player.env.task._demo_env_offset = player.env.task._init_random_dir_env_indexes(start_at = 0)
        player.env.task._random_dir_speed_mimic_upper_prob = 0
        player.env.task._init_random_dir_upper_env_indexes()
        player._compose_demo_targets_share = 0
        player.env.task._init_compose_env_indexes(start_at = player.env.task._demo_env_offset)

        player.env.task._start_state_motion_weights[:] = 0
        player.env.task._start_state_motion_weights[demo_idx_pool] = 1/len(demo_idx_pool)
        player.env.task._state_init = HumanoidAMPRobust.StateInit.Start
        player.env.task._demo_init = HumanoidAMPRobust.StateInit.Start
        player.env.task._state_init_rotate = False
        player.env.task._state_demo_rotate = False
        player.env.task._fall_init_prob = 0
        player.env.task._start_demo_at_agent_state_prob = 0
        player.env.task._uniform_targets = True
        player.env.task._switch_demos_within_episode = False

        player.env.task._random_dir_speed_env_idxs = torch.zeros([0]).type(torch.LongTensor).cuda()
        player.env.task.show_traj = False
        player.env.task.show_root_traj = True
        player.env.task.show_headings = False

        player._render_color_for_speed = False

        player.env.task._enable_lk_jl_mask = False
        player.env.task._jl_mask_prob = 0
        player.env.task._enable_lookahead_mask = True
        # player.env.task.lookahead_mask_pool_sample_weights[:] = 0 
        # player.env.task.lookahead_mask_pool_sample_weights[3] = 1 # Global Lookahead
        player.env.task._use_predefined_lookahead_mask = True
        player.env.task._predefined_lookahead_mask = torch.ones(player.env.task._lookahead_obs_dim).type(torch.bool).cuda()
        player.env.task._predefined_lookahead_mask[-51:] = mask.repeat_interleave(3)

        player.env_reset(torch.arange(player.env.task.num_envs).cuda())
        ###################################################################################################################

        player.env.task.max_episode_length = torch.max(player.env.task._motion_max_steps[demo_idx_pool]) - 10
        player.games_num = 1
        if collect_dataset:
            buffer = player.collect_dataset(player.env.task.max_episode_length)
        else:
            player.run()
            buffer = None
    restore_env_vars_from_cache(player.env.task, cached_vars)

    return buffer
`
But why is the invocation like this?Their number of parameters does not match.
Thank you.
